llvm/lib/Target/AMDGPU/GCNHazardRecognizer.cpp

```diff
@@ -858,9 +858,12 @@ int GCNHazardRecognizer::createsVALUHazard(const MachineInstr &MI) {
   }
 
   if (TII->isFLAT(MI)) {
```

Comment:
Is the MIMG case above doing anything? It looks like it's just an assert?  I'm also surprised we're querying the operand before the instruction type

Reply:
> Is the MIMG case above doing anything? It looks like it's just an assert?

Yes, seems that way. The comment suggests the code exists more as a reminder to include the assert than check anything - 

`All our MIMG definitions use a 256-bit T#, so we can skip checking for them.`